tools/update-distgit.py

```diff
@@ -0,0 +1,124 @@
+#!/usr/bin/python3
+import argparse
+import contextlib
+import os
+import subprocess
+import tempfile
+import urllib.request
+from datetime import datetime
+
+
+def read_file(path) -> str:
+    """
+    Return the content of the file on the given path.
+    """
+    with open(path) as f:
+        return f.read()
+
+
+def download_github_release_tarball(user: str, project: str, version: str) -> str:
+    """
+    Download a github release tarball of the given project.
+    """
+    project_url = f"https://github.com/{user}/{project}"
+    tarball_url = f"{project_url}/archive/v{version}.tar.gz"
+
+    local_path = f"{project}-{version}.tar.gz"
+    urllib.request.urlretrieve(
+        tarball_url,
+        local_path
+    )
+
+    return local_path
+
+
+@contextlib.contextmanager
+def extracted_tarball(path: str):
+    """
+    Extract a tarball into a temporary directory.
+    Yields the temporary directory.
+    """
+    with tempfile.TemporaryDirectory() as tempdir:
+        subprocess.run(["tar", "-xf", path, "--directory", tempdir], check=True)
+        yield tempdir
+
+
+def merge_specfiles(upstream: str, downstream: str, version: str, author: str):
+    """
+    Merge the upstream specfile with the changelog from downstream and add a new
+    changelog entry.
+    """
+    upstream_spec_lines = upstream.splitlines()
+    downstream_spec_lines = downstream.splitlines()
+
+    # Find where changelog starts in both specfiles
+    changelog_start_in_up_spec = upstream_spec_lines.index("%changelog")
+    changelog_start_in_down_spec = downstream_spec_lines.index("%changelog")
+
+    # Create a new changelog entry
+    date = datetime.now().strftime("%a %b %d %Y")
+    changelog = f"""\
+* {date} {author} - {version}-1
```

Comment:
Since we don't handle rebuilds (`{version}-N`) in the script, maybe it should be mentioned in the guide that rebuilds (though rare) follow a different flow.  At least as a sidenote that we shouldn't look to the guide if we're doing a rebuild release.

Reply:
From my point of view, the guide is only for releasing a new upstream version and getting it into downstreams. Doing rebuilds is a topic that's relevant only for downstream and I think it doesn't make much sense to even mention it in upstream guide... In other words, we cannot write here everything about RPM packaging and I suggest drawing the line before rebuilds, they are rare and not advised.